core-bundle/src/Resources/contao/library/Contao/FeedItem.php

```diff
@@ -113,12 +118,29 @@ public function addEnclosure($strFile, $strUrl=null, $strMedia='enclosure')
 			$strUrl = Environment::get('base');
 		}
 
+		$fileUrl = $strUrl . System::urlEncode($strFile);
 		$objFile = new File($strFile);
 
+		$size = StringUtil::deserialize($size, true);
+
+		if ($size && $objFile->isImage)
+		{
+			$image = System::getContainer()->get('contao.image.image_factory')->create($rootDir . '/' . $strFile, $size);
+
+			if ($image instanceof DeferredImageInterface)
+			{
+				System::getContainer()->get('contao.image.resizer')->resizeDeferredImage($image);
```

Comment:
why do we need to force-resize deferred images? They are not embedded in the feed so they would just be created on request?

Reply:
True, this is probably specific to my original use case. We used the RSS feed to import news articles into and send them with CleverReach. Their client did not wait long enough for the images to be created and just did not load them at all. 

We could make DeferredImages the default, but I would like to add a checkbox then, that allows me to create images on feed creation. WDYT?

Reply:
@ausi /cc

Reply:
if you have that specific problem, you should configure that in your application – disable deferred images or run the image-generation command in a cron job.

Reply:
> but I would like to add a checkbox then, that allows me to create images on feed creation. WDYT?

I would not add this checkbox as this requirement is a quite specific use-case IMO.

But we should make sure that your use-case is possible at least. Is there an event or hook that you can use to intercept the feed contents? If so, it might be possible to resize the images like the newsletter module does with `$objFile->exists() || $objFile->createIfDeferred()`: https://github.com/contao/contao/blob/4299a5116b73cdbcd024c2d86b0533ae15ec3137/core-bundle/src/Resources/contao/library/Contao/Email.php#L476

Reply:
Agreed, I will look into that. As far as I see, there is no Hook/Event yet, so we would have to add one. I will figure out, where would be the best place.

Reply:
Alternatively: If we migrate the feed composition to templates, this would be easily adjustable for users as well.

Reply:
> Alternatively: If we migrate the feed composition to templates, this would be easily adjustable for users as well.

not related to this PR, and something we might look at when rewriting to a controller 😉